marklogic-data-hub/src/main/java/com/marklogic/hub/job/impl/JobMonitorImpl.java

```diff
@@ -54,6 +54,10 @@ public void setupClient() {
         batches = new Batches(client);
     }
 
+    public void setHubConfig(HubConfig hubConfig) {
```

Comment:
Instead of a setter, let's add a no-arg constructor and then a constructor that requires a HubConfig. The client can then create a new JobMonitorImpl based on a HubConfig.